docs/tenant.puml

```diff
@@ -0,0 +1,185 @@
+    ]
+    agent es_k [
+        ==elasticsearch
+        this is an elasticsearch instance
+        that is neither a data node nor
+        a master. It's probably just here
+        to make addressing easier for kibana
+    ]
+    agent routeregistrar [
+        == route registrar
+        tells nats where kibana is
+        so gorouter can route traffic
+    ]
+    kbn -r-> es_k
+    kbn -l-> kibanaplugin
+
+}
+
+node "redis (n x environment)" as redis_host {
+    agent redis [
+        ==redis
+    ]
+}
+
+node "archiver (N x environment)" as archiver {
+    agent ingestor_firehose_a [
+        ==ingestor_firehose-cloudfoundry
+        pulls logs from doppler
+    ]
+
+    agent ingestor_syslog [
+        ==ingestor_syslog
+        logstash process
+        listens for syslog messages,
+        writes docs raw to S3
+    ]
+}
+
+node "ingestor (N x environment)" as ingestor {
+    agent ingestor_firehose_i [
+        ==ingestor_firehose-cloudfoundry
+        pulls logs from doppler
+    ]
+
+    agent ingestor_syslog [
+        ==ingestor_syslog
+        logstash process
+        listens for syslog messages,
+        does some minimal parsing
+        writes docs to elasticsearch
+    ]
+    agent es_i [
+        ==elasticsearch
+        this elasticsearch instance
+        is neither a data node nor
+        a master. It's probably just here
+        to make addressing easier for ingestor_syslog
+    ]
+    ingestor_syslog -> es_i
+}
+
+node "elasticsearch_master (3 x environment)" as elasticsearch_master {
+    agent es_em [
+        ==elasticsearch
+        master nodes are used
+        for coordination only
+    ]
+}
+
+node "elasticsearch_data (N x environment)" as elasticsearch_data {
+    agent es_ed [
+        ==elasticsearch
+        data nodes are where the
+        logs are actually stored
+    ]
+}
+
+
+node "maintenance (1 x environment)" as maintenance {
+    agent curator [
+        ==curator
+        curator prunes indices based on name
+        it's capable of other maintenance tasks
+        but we don't leverage them
+    ]
+    agent elasticsearch_exporter [
+        ==elasticsearch_exporter
+        gathers stats about elasticsearch
+        exports them to prometheus
+    ]
+    agent kibana_objects [
+        ==upload-kibana-objects
+        uploads index templates, 
+        vizualizations, etc
+    ]
+    agent es_m [
+        ==elasticsearch
+        this elasticsearch instance
+        is neither a data node nor
+        a master. It's probably just here
+        to make addressing easier for curator
+    ]
+    curator -> es_m
+    kibana_objects -u-> kbn
+    elasticsearch_exporter -r-> es_m
+}
+}
+
+user --> ALB
+user --> UAAALB
+ALB --> gorouter
+UAAALB --> uaa
+es_k -down-> es_em
+es_i -down-> es_em
+es_m -up-> es_em
+es_em <-down-> es_ed
+kibanaplugin -u-> UAAALB : authenticate users
+kibanaplugin -u-> capi : get user roles
+kibanaplugin -d-> redis : share state\nbetween instances
```

Comment:
If `kibanaplugin == kibana-cf-auth-proxy`, then we're not actually using Redis to share state between instances but cryptographically signed cookies, correct?

Reply:
right - these diagrams are about the current state of the logsearch stack and don't consider the desired state of the opensearch stack